mono/metadata/object.c

```diff
@@ -7281,16 +7280,15 @@ mono_string_get_pinned (MonoStringHandle str, MonoError *error)
 		mono_error_set_out_of_memory (error, "Could not allocate %" G_GSIZE_FORMAT " bytes", size);
 		return news;
 	}
-	const gboolean pin = FALSE; // FIXME pin is slow and breaks profiler test.
-	gchandle_t str_gchandle = 0;
-	gchandle_t new_gchandle = 0;
-	memcpy (pin ? mono_string_handle_pin_chars (news, &new_gchandle)
-		    : mono_string_chars_internal (MONO_HANDLE_RAW (news)),
-		pin ? mono_string_handle_pin_chars (str, &str_gchandle)
-		    : mono_string_chars_internal (MONO_HANDLE_RAW (str)),
+
+	MONO_ENTER_NO_SAFEPOINTS;
```

Comment:
Static analysis needs to learn about this at some point, but plenty to do before then.

Reply:
Actually it might just be suppress within the macro (and a scope). i.e. already  have trivial to  use building blocks.